smart-contracts/contracts/cl-vault/README.md

```diff
@@ -155,3 +155,41 @@ $$K=\sqrt{P_c}\sqrt{P_l}\frac{\sqrt{P_u}-\sqrt{P_{c}}}{\sqrt{P_{c}}-\sqrt{P_l}}$
 This current math assumes an ideal pool with infinite liquidity. In reality, we will need to account for slippage.
 
 We can actually create further optimizations here by increasing the amount of tokens to swap by half of the expected slippage, which will lead to returning (not being able to deposit) less tokens.
+
+# Multi-position support
+## intro
+The CL vault supports multiple positions internally. 
+
+### position ratios
+Each position is saved in the contract with a ratio. This ratio dictates how many tokens are sent to a position when a user deposits.
+
+Given any current spot price, each position needs a different amount of tokens. Lets take 3 positions $P_1$, $P_2$ and $P_3$. at some price $P$, $P_1$ needs 30% of token0 and 70% of token1. $P_2$ needs 45% token0 and 55% token1 and $P_3$ 80% token0 and 20% token1. Effectively this means that each position has the following internal ratio:
```

Comment:
@LaurensKubat  I think, an example with real tokens example with values would give better visualization. 
Like uosmo as token0 and ibc uatom as token 1, And deposit is 100000000uosmo, and 500000000 ibc atom, then what will be the internal ratio, external ratio, actual ratio and effective ratio.